labelbox/schema/dataset.py

```diff
@@ -260,11 +260,10 @@ def upload_if_necessary(item):
             row_data = item['row_data']
             if os.path.exists(row_data):
                 item_url = self.client.upload_file(item['row_data'])
-                item = {
-                    "row_data": item_url,
-                    "external_id": item.get('external_id', item['row_data']),
-                    "attachments": item.get('attachments', [])
-                }
+                item['row_data'] = item_url
+                if 'external_id' not in item:
+                    # Default `external_id` to local file name
+                    item['external_id'] = row_data
```

Comment:
How is this different?

Reply:
it uses the original `item`, and alters it, rather than constructing a brand new `item`

Reply:
What is the need for this?

Reply:
oh sorry, I should have read the description.

Reply:
No worries, it's a subtle change haha